### DISCUSSION TITLE
C# Markup - Initial binding in Windows fails.

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#15618] [Q&A]
I was just playing with C# Markup and MVVM, and I have noticed that the binding of `vm.Title` works fine in the Skia head, but it doesn't for Windows. (See images below)

Maybe I am doing something wrong. I've added a minimal example project.

**MainPage.cs**
```
public sealed partial class MainPage : Page
{
    public MainPage() {
        this.Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
            .DataContext(new MainPageViewModel(), (page, vm) => page
            .Content(
                new FrameControl().Caption(() => vm.Title))
                );
    }
}
```
**MainPageViewModel.cs**
```
public partial class MainPageViewModel : ObservableObject
{
    [ObservableProperty]
    private string _title = "Hello Uno Platform!";
}
```
**FrameControl.cs**
```
public partial class FrameControl : UserControl
{
    public string Caption {
        get { return (string)GetValue(CaptionProperty); }
        set { SetValue(CaptionProperty, value); }
    }

    public static readonly DependencyProperty CaptionProperty =
        DependencyProperty.Register(nameof(Caption), typeof(string), typeof(FrameControl), new PropertyMetadata(""));

    public FrameControl() {
        this.Content(
            new Border()
                .Child(
                    new Grid().ColumnDefinitions("Auto, *")
                        .Children(
                            new Border().Grid(column: 0).Width(300).Background(OsmoBrush.OsmoGradientBlau),
                            new TextBlock().Grid(column: 1).Text(x => x.Bind(() => this.Caption).Source(this).Mode(BindingMode.OneWay)).FontSize(30).Foreground("#ffffff"))));
    }
}
```
![osmouiskia](https://github.com/unoplatform/uno/assets/24267381/dd79dc42-3b88-4043-8567-53b2824c7885)

![osmouiwin](https://github.com/unoplatform/uno/assets/24267381/14e80b1d-1f9f-49d6-a19c-458d38ab0335)

[osmoui2.zip](https://github.com/unoplatform/uno/files/14392342/osmoui2.zip)

**Comment by dansiegel** [comment]
Hi @mcNets thanks for reporting this. I've raised this to the team and we are looking into it. There isn't anything that I can see that you're doing wrong. The code looks right to me. We will need to determine if this is an issue in WinUI itself or if there is something we need to adjust in C# Markup so unfortunately I can't give you much more of an update than that.

**Reply by erikvilima** [reply]
@mcNets @dansiegel Binding also fails on a WinUI app without Uno / C# Markup. Related issue to track: https://github.com/microsoft/WindowsAppSDK/issues/4237
